### PR TITLE
Send the correct JWT URLs to cross device client

### PR DESCRIPTION
**Pull request by kopijunkie**
# Problem
Currently, the JS SDK is not sending the URLs to the cross device client, so the default ones are used instead of the ones in the JWT.

Additionally id.onfido.com cross-device link does not work if the desktop SDK was configured with a US region JWT.

# Solution
* Cross device client should use the URLs inside the JWT
* Add UI test that fetches a US token via a US token factory and runs through the happy path for the x-device flow

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [n/a] Has the README been updated?
- [n/a] Has the RELEASE_GUIDELINES been updated?
- [n/a] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [n/a] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [x] Have any new automated tests been implemented or the existing ones changed?
- [n/a] Have any new manual tests been written down or the existing ones changed?
- [n/a] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

File: src/demo/demo.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ if (process.env.NODE_ENV === 'development') {
 }
 
 const getToken = (hasPreview, onSuccess) => {
-  const url = process.env.JWT_FACTORY
+  const region = queryParamToValueString.region
+  const url = region === 'US' ? process.env.US_JWT_FACTORY : process.env.JWT_FACTORY
+  console.log('* JWT Factory URL:',url,'for',region,process.env.NODE_ENV)
```
Comment (joninvski) [inline src/demo/demo.js]: missing whistespace after `,`

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

File: webpack.config.babel.js [hunk 2]
```diff
@@ -72,23 +72,28 @@ const baseStyleLoaders = (modules, withSourceMap) => [
...
     [rule]: [`${__dirname}/node_modules`],
-    use:
-     [
-       disableExtractToFile || !PRODUCTION_BUILD ?
-         'style-loader' : MiniCssExtractPlugin.loader,
-       ...baseStyleLoaders(modules, withSourceMap)
-     ]
+    use: [
+      disableExtractToFile || !PRODUCTION_BUILD
+        ? 'style-loader'
```
Comment (stefaniacardenas) [inline webpack.config.babel.js]: Minor. I understand this has been automatically prettified, but everywhere else in the code we use the following style:
```
conditionMet ?
  doSomthing :
  doSomethingElse
```

We need to agree on a stricter style guide

Comment (kopijunkie) [reply webpack.config.babel.js]: Will update. But yeah, the `.prettierrc` file I added is a basic one & it should follow whatever is in our `.eslintrc` for most linting but until we agree on that for now there definitely are instances (like this one) where the default conflicts with what we have.

Comment (kopijunkie) [reply webpack.config.babel.js]: I personally prefer the style we're currently using actually.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Confirm/Confirm.js [hunk 3]
```diff
@@ -94,6 +94,7 @@ class Confirm extends Component {
   uploadCaptureToOnfido = () => {
     const { urls, capture, method, side, token, poaDocumentType, language } = this.props
     const url = urls.onfido_api_url
+    console.log('Upload Capture API URL:',url)
```
Comment (stefaniacardenas) [inline src/components/Confirm/Confirm.js]: Did you intend to leave this here?

File: test/specs/scenarios/crossDevice.js [hunk 4]
```diff
@@ -268,5 +268,21 @@ export const crossDeviceScenarios = async (lang) => {
...
+    it('should complete cross device e2e flow with a US JWT', async () => {
+      driver.get(`${baseUrl}&region=US`)
+      welcome.primaryBtn().click()
+      documentSelector.passportIcon.click()
+      runThroughCrossDeviceFlow()
+      uploadFileAndClickConfirmButton(documentUpload, confirm, 'passport.jpg')
+      selfieIntro.clickOnContinueButton()
+      camera.takeSelfie()
+      confirm.confirmBtn().click()
+      crossDeviceClientSuccess.verifyUIElements(copy)
```
Comment (stefaniacardenas) [inline test/specs/scenarios/crossDevice.js]: This doesn't actually test that cross device is calling a US endpoint. As part of another story we can detect the network request using something like [browsermob-proxy](https://github.com/lightbody/browsermob-proxy) to make sure that the request fired is calling the right endpoint.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/App/index.js [hunk 5]
```diff
@@ -83,13 +86,38 @@ class ModalApp extends Component {
         this.props.actions.setIdDocumentType(enabledDocs[0])
       }
     }
+    if (!options.mobileFlow && token && token !== prevToken) {
+      const jwtUrls = fetchUrlsFromJWT(token)
```
Comment (josokinas) [inline src/components/App/index.js]: [Minor] Odd function naming `fetchUrlsFromJWT` - initially I thought that should be an async call to some BE so here async/await would be needed, but then looked up at the function... IMO it should be `getUrlsFromJWT` or something similar.

File: src/components/ReduxAppWrapper/store/reducers/globals.js [hunk 6]
```diff
@@ -11,6 +11,13 @@ const initialState = {
   isNavigationDisabled: false,
   isFullScreen: false,
   deviceHasCameraSupport: false,
+  urls: {
```
Comment (josokinas) [inline src/components/ReduxAppWrapper/store/reducers/globals.js]: [Minor] Not sure if having mapping with not consistent naming is good - in the future you might get lost in which environment variable is what on the app.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-915-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/ReduxAppWrapper/store/reducers/globals.js (hunk 6 above)
Comment (kopijunkie) [reply src/components/ReduxAppWrapper/store/reducers/globals.js]: Good point! Will discuss with team to agree on a consistent naming.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-915-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://915-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**